src/main/java/eu/europeana/metis/sandbox/service/workflow/MediaProcessingServiceImpl.java

```diff
@@ -22,75 +19,93 @@
 import eu.europeana.metis.sandbox.domain.RecordError;
 import eu.europeana.metis.sandbox.domain.RecordInfo;
 import eu.europeana.metis.sandbox.service.util.ThumbnailStoreService;
-import java.io.IOException;
 import java.util.LinkedList;
 import java.util.List;
 import org.apache.commons.collections.CollectionUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 @Service
 class MediaProcessingServiceImpl implements MediaProcessingService {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(MediaProcessingServiceImpl.class);
 
-  private final RdfConverterFactory converterFactory;
-  private final MediaProcessorFactory processorFactory;
   private final ThumbnailStoreService thumbnailStoreService;
+  private final MediaExtractor mediaExtractor;
+  private final RdfSerializer rdfSerializer;
+  private final RdfDeserializer rdfDeserializer;
 
   public MediaProcessingServiceImpl(
-      RdfConverterFactory converterFactory,
-      MediaProcessorFactory processorFactory,
-      ThumbnailStoreService thumbnailStoreService) {
-    this.converterFactory = converterFactory;
-    this.processorFactory = processorFactory;
+      ThumbnailStoreService thumbnailStoreService,
+      @Qualifier("rdfSerializer")
```

Comment:
The qualifiers are probably not required since they are exactly identical to the name of the field.

Reply:
yes. correct. removed. done!